### PR TITLE
libs version update to support latest android studio

### PR DESCRIPTION
**Pull request by parmeshtoyou**
Hi Lukas,
After purchasing the course from udemy, i tried building your branch but it was not succeeded due to Android studio version update.
I have updated required libraries, gradle version and able to build the code on latest android version.

Android Studio Ladybug Feature Drop | 2024.2.2 Canary 4
Build #AI-242.23339.11.2422.12417746, built on September 26, 2024

Please review the changes and we can either keep these code in a new branch or update the existing one.

This will save other new devs time.

Thanks for this course.

### CODE REVIEW

**Comment by cavega** [comment]
@parmeshtoyou I ran into this same issue yesterday and updated the tools in my local branch.

@LukasLechnerDev It would be very helpful for future students to be able to clone the project and be up-and-running building the project. Would you be able to review this PR created by @parmeshtoyou? Thanks.

File: app/src/main/java/com/lukaslechner/coroutineusecasesonandroid/usecases/coroutines/usecase1/PerformSingleNetworkRequestViewModel.kt [hunk 1]
```diff
@@ -8,6 +8,6 @@ class PerformSingleNetworkRequestViewModel(
 ) : BaseViewModel<UiState>() {
 
     fun performSingleNetworkRequest() {
-
+        uiState.value = UiState.Error("something went wrong!")
```
Comment (LukasLechnerDev) [inline app/src/main/java/com/lukaslechner/coroutineusecasesonandroid/usecases/coroutines/usecase1/PerformSingleNetworkRequestViewModel.kt:11]: I guess this doesn't belong here?

**Comment by LukasLechnerDev** [comment]
Thank you @parmeshtoyou for your contribution 💪

Could you please: 
- Address the one comment that I wrote
- merge master into this MR (I fixed the pipeline)

Then I think we can merge it!

**Comment by parmeshtoyou** [comment]
@LukasLechnerDev can we merge this PR? Provided comment addressed.

**Comment by LukasLechnerDev** [comment]
Its merged! Thank you so much for your effort @parmeshtoyou

**Pull request merged**